sdk/storage/src/shares/share_file_client.cpp

```diff
@@ -0,0 +1,877 @@
+  }
+
+  Azure::Core::Response<CreateFileResult> FileClient::Create(
+      int64_t fileSize,
+      const CreateFileOptions& options) const
+  {
+    auto protocolLayerOptions = ShareRestClient::File::CreateOptions();
+    protocolLayerOptions.Metadata = options.Metadata;
+    protocolLayerOptions.FileAttributes = FileAttributesToString(options.SmbProperties.Attributes);
+    if (protocolLayerOptions.FileAttributes.empty())
+    {
+      protocolLayerOptions.FileAttributes = FileAttributesToString(FileAttributes::None);
+    }
+    protocolLayerOptions.FileCreationTime = options.SmbProperties.FileCreationTime;
+    protocolLayerOptions.FileLastWriteTime = options.SmbProperties.FileLastWriteTime;
+    if (options.FilePermission.HasValue())
+    {
+      protocolLayerOptions.FilePermission = options.FilePermission.GetValue();
+    }
+    else if (!options.SmbProperties.FilePermissionKey.empty())
+    {
+      protocolLayerOptions.FilePermissionKey = options.SmbProperties.FilePermissionKey;
+    }
+    else
+    {
+      protocolLayerOptions.FilePermission = "inherit";
+    }
+    protocolLayerOptions.XMsContentLength = fileSize;
+    if (options.HttpHeaders.HasValue())
+    {
+      if (!options.HttpHeaders.GetValue().ContentType.empty())
+      {
+        protocolLayerOptions.FileContentType = options.HttpHeaders.GetValue().ContentType;
+      }
+      if (!options.HttpHeaders.GetValue().ContentEncoding.empty())
+      {
+        protocolLayerOptions.FileContentEncoding = options.HttpHeaders.GetValue().ContentEncoding;
+      }
+      if (!options.HttpHeaders.GetValue().ContentLanguage.empty())
+      {
+        protocolLayerOptions.FileContentLanguage = options.HttpHeaders.GetValue().ContentLanguage;
+      }
+      if (!options.HttpHeaders.GetValue().CacheControl.empty())
+      {
+        protocolLayerOptions.FileCacheControl = options.HttpHeaders.GetValue().CacheControl;
+      }
+      if (!options.HttpHeaders.GetValue().ContentDisposition.empty())
+      {
+        protocolLayerOptions.FileContentDisposition
+            = options.HttpHeaders.GetValue().ContentDisposition;
+      }
+    }
+    protocolLayerOptions.FileContentMD5 = options.FileContentMD5;
+    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
+    return ShareRestClient::File::Create(
+        m_shareFileUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  Azure::Core::Response<DeleteFileResult> FileClient::Delete(const DeleteFileOptions& options) const
+  {
+    auto protocolLayerOptions = ShareRestClient::File::DeleteOptions();
+    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
+    return ShareRestClient::File::Delete(
+        m_shareFileUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
+  }
+
+  Azure::Core::Response<DownloadFileResult> FileClient::Download(
+      const DownloadFileOptions& options) const
+  {
```

Comment:
reliable stream

Reply:
so either do it in this PR or create an issue to track, in case we'd forget

Reply:
Implemented.